### PR TITLE
Add skeleton for frame analysis feature

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Work towards https://github.com/flutter/devtools/issues/3485.
frame-analysis-gif

This work is hidden behind a flag, which will be turned on once the feature is complete.

The general workflow is as such:
- in the flutter frames chart, frames with jank are marked with an icon suggesting more analysis is available (right now, marked only for UI jank until we have more actionable data from the raster thread)
- When this frame is selected, a button with the same icon and tooltip "Analyze selected frame" will appear in the Timeline Events section header.
- Clicking this button will open a tab for this frame's analysis

Some other items to note:
- Tabs will remain open until manually closed.
- Selecting a new frame will open the timeline frame chart to that frame's time range (current behavior)

### CODE REVIEW

File: packages/devtools_app/lib/src/common_widgets.dart [hunk 1]
```diff
@@ -793,15 +795,37 @@ class AreaPaneHeader extends StatelessWidget implements PreferredSizeWidget {
...
 
+  List<Widget> _buildActions() {
+    return [
+      if (scrollableCenterActions.isEmpty)
+        Expanded(
+          child: Row(
+            children: leftActions,
+          ),
+        ),
+      if (scrollableCenterActions.isNotEmpty) ...[
```
Comment (elliette) [inline packages/devtools_app/lib/src/common_widgets.dart:813]: More of a Dart clarification question for me - why do `...leftActions` etc. need to be enclosed in a list if that list is spread?

File: packages/devtools_app/lib/src/performance/timeline_flame_chart.dart [hunk 2]
```diff
@@ -42,89 +45,94 @@ class TimelineFlameChartContainer extends StatefulWidget {
...
+        content = LayoutBuilder(
+          builder: (context, constraints) {
+            return TimelineFlameChart(
+              controller.data,
+              width: constraints.maxWidth,
+              height: constraints.maxHeight,
+              selectionNotifier: controller.selectedTimelineEvent,
+              searchMatchesNotifier: controller.searchMatches,
+              activeSearchMatchNotifier: controller.activeSearchMatch,
+              onDataSelected: (e) => controller.selectTimelineEvent(e),
```
Comment (elliette) [inline packages/devtools_app/lib/src/performance/timeline_flame_chart.dart:117]: Can this just be `onDataSelected: controller.selectTimelineEvent` ?

**Review by elliette — APPROVED** [review]

File: packages/devtools_app/lib/src/performance/timeline_flame_chart.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/performance/timeline_flame_chart.dart:117]: yep - done

File: packages/devtools_app/lib/src/common_widgets.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/common_widgets.dart:813]: Take this scenario:
leftActions = [action1, action2]
scrollableCenterAcitons = [action3, action4]
rightActions = [action5, action6].

Breaking down this method in it's current state with the spread operator, `_buildActions` would return this `List<Widget>`:
```
<Widget>[
  ...[action1, action2,]
  Exanded(ListView(children[action3, action4])),
  ...[action5, action6]
] 
```
which turns into
```
<Widget>[
  action1,
  action2,
  Exanded(child: ListView(children: [action3, action4])),
  action5,
  action6,
] 
```

Without the spread operator on `leftActions`, this is what `_buildActions` would return:
```
<Widget>[
  [action1, action2,] // This is not of type Widget - it is of type List<Widget>, which gives a compilation error
  Exanded(child: ListView(children: [action3, action4])),
  ...[action5, action6]
] 
```

**Pull request merged**